clients/go/http_test.go

```
@@ -0,0 +1,528 @@
+// token, on a same-host hop) to whatever Location names.
+func TestRESTRedirectsWithCredentials(t *testing.T) {
+	// The redirect target records anything that reaches it. Nothing should.
+	var leaked http.Header
+	dst := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		leaked = r.Header.Clone()
+		_, _ = io.WriteString(w, `[]`)
+	}))
+	defer dst.Close()
+
+	redirector := func(w http.ResponseWriter, r *http.Request) {
+		http.Redirect(w, r, dst.URL+r.URL.Path, http.StatusFound)
```

golangci-lint (2.12.2)

[medium] G710: Open redirect via taint analysis

(gosec)